src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.FormatDocument.cs

```diff
@@ -34,15 +64,54 @@ private bool TryExecuteCommand(FormatDocumentCommandArgs args, CommandExecutionC
                 return false;
             }
 
-            var formattingService = document.GetLanguageService<IEditorFormattingService>();
-            if (formattingService == null || !formattingService.SupportsFormatDocument)
+            return ExecuteCommandAsync(document, args.TextView, context).Result;
```

Comment:
this is not good for several reasons.  
1. you are blocking on async work. 
2. if the async work throws an exception (like cancellation), this will become an aggregate exception.
3. it's unclear ot me how you can show progress if you do this.

Reply:
fixed